CHANGELOG.md

```diff
@@ -7,6 +7,7 @@
 * <How to hit and notice issue? what was the impact?> ([#????](https://github.com/realm/realm-core/issues/????), since v?.?.?)
 * Performance regression for some scenarios of writing/creating objects 
   with a primary key. ([#4522](https://github.com/realm/realm-core/issues/4522))
+* Setting a collection with a nullable value type to null would hit an assertion failure instead of clearing the collection.
```

Comment:
This was a pre-monorepo object store bug so there's no version that it was introduced in.